### PR TITLE
Some rovinski cleanup rebased

### PR DESCRIPTION
**Pull request by oharboe**
Some commits that rebased cleanly from #1533

### CODE REVIEW

File: src/drt/src/dr/FlexDR.cpp [hunk 1]
```diff
@@ -322,18 +322,8 @@ void FlexDR::initGCell2BoundaryPin()
...
-              } else {
-                hasLeftBound = false;
-              }
-              if (ep.x() >= rightBound) {
-                hasRightBound = true;
-              } else {
-                hasRightBound = false;
-              }
+              bool hasLeftBound = bp.x() < leftBound;
+              bool hasRightBound = ep.x() >= rightBound;
```
Comment (maliberty) [inline src/drt/src/dr/FlexDR.cpp:326]: these can be const now

File: src/drt/src/dr/FlexDR.cpp [hunk 2]
```diff
@@ -354,18 +344,8 @@ void FlexDR::initGCell2BoundaryPin()
...
-              } else {
-                hasBottomBound = false;
-              }
-              if (ep.y() >= topBound) {
-                hasTopBound = true;
-              } else {
-                hasTopBound = false;
-              }
+              bool hasBottomBound = bp.y() < bottomBound;
+              bool hasTopBound = ep.y() >= topBound;
```
Comment (maliberty) [inline src/drt/src/dr/FlexDR.cpp:348]: const

File: src/drt/src/dr/FlexDR_init.cpp [hunk 3]
```diff
@@ -2542,6 +2535,7 @@ void FlexDRWorker::route_queue_init_queue(queue<RouteQueueEntry>& rerouteQueue)
     // ripup all nets and clear objs here
     // nets are ripped up during initNets()
     vector<drNet*> ripupNets;
+    ripupNets.reserve(nets_.size());
     for (auto& net : nets_) {
       ripupNets.push_back(net.get());
     }
```
Comment (maliberty) [inline src/drt/src/dr/FlexDR_init.cpp:2541]: This can just be

vector<drNet*> ripupNets(nets_);

File: src/drt/src/dr/FlexGridGraph.cpp [hunk 4]
```diff
@@ -66,20 +69,23 @@ void FlexGridGraph::initGrids(
   // initialize all grids
   frMIdx xDim, yDim, zDim;
   getDim(xDim, yDim, zDim);
+  int capacity = xDim * yDim * zDim;
```
Comment (maliberty) [inline src/drt/src/dr/FlexGridGraph.cpp:72]: const

**Review by maliberty — APPROVED** [review]
After a review it is helpful if you don't force push, that way I can use the 'changes since last review' feature.

**Comment by oharboe** [comment]
> After a review it is helpful if you don't force push, that way I can use the 'changes since last review' feature.

Ah. I'll leave the ```git commit --fixup```s in place next time.

**Comment by maliberty** [comment]
It looks like these are subtly different in a way I didn't notice:
```
/OpenROAD/src/drt/src/dr/FlexDR_init.cpp:2537:35: error: no matching function for call to 'std::vector<fr::drNet*>::vector(std::vector<std::unique_ptr<fr::drNet> >&)'
```

I guess we'll have to go back to the prior version.  Sorry for the hassle.

**Pull request merged**